OptimizelySDK/Optimizely.cs

```diff
@@ -711,6 +712,140 @@ public OptimizelyJSON GetFeatureVariableJSON(string featureKey, string variableK
+            }
+            var variableMap = new Dictionary<string, object>();
+            if (flag?.Variables != null && !allOptions.Contains(OptimizelyDecideOption.EXCLUDE_VARIABLES))
+            {
+
+                foreach (var featureVariable in flag?.Variables)
+                {
+                    string variableValue = featureVariable.DefaultValue;
+                    if (featureEnabled)
+                    {
+                        var featureVariableUsageInstance = variation.GetFeatureVariableUsageFromId(featureVariable.Id);
+                        if (featureVariableUsageInstance != null)
+                        {
+                            variableValue = featureVariableUsageInstance.Value;
+                        }
+                    }
+
+                    var typeCastedValue = GetTypeCastedVariableValue(variableValue, featureVariable.Type);
+
+                    if (typeCastedValue is OptimizelyJSON)
+                        typeCastedValue = ((OptimizelyJSON)typeCastedValue).ToDictionary();
+
+                    variableMap.Add(featureVariable.Key, typeCastedValue);
+                }
+            }
+            
+            var optimizelyJSON = new OptimizelyJSON(variableMap, ErrorHandler, Logger);
+
+            var decisionSource = flagDecision?.Source ?? FeatureDecision.DECISION_SOURCE_ROLLOUT;
+            if (!allOptions.Contains(OptimizelyDecideOption.DISABLE_DECISION_EVENT))
+            {
+                SendImpressionEvent(flagDecision.Experiment, variation, userId, userAttributes, config, key, decisionSource, featureEnabled);
+                decisionEventDispatched = true;
+            }
+            var reasonsToReport = decisionReasons.ToReport();
+            var variationKey = flagDecision.Variation?.Key;
+
+            // TODO: add ruleKey values when available later. use a copy of experimentKey until then.
+            var ruleKey = flagDecision.Experiment?.Key;
+
+            var decisionInfo = new Dictionary<string, object>
+            {
+                { "flagKey", key },
+                { "enabled", featureEnabled },
+                { "variables", variableMap },
+                { "variationKey", variationKey },
+                { "ruleKey", ruleKey },
+                { "reasons", decisionReasons },
+                { "decisionEventDispatched", decisionEventDispatched },
+                { "featureEnabled", featureEnabled },
+            };
+
+            NotificationCenter.SendNotifications(NotificationCenter.NotificationType.Decision, DecisionNotificationTypes.FEATURE, userId,
+               userAttributes ?? new UserAttributes(), decisionInfo);
+
+            return new OptimizelyDecision(
+                variationKey,
+                featureEnabled,
+                optimizelyJSON,
+                ruleKey,
+                key,
+                user,
+                reasonsToReport.ToArray());
```

Comment:
It looks like  OptimizelyDecision returns an array of strings instead of a list.
defaultDecideOptions is set as an array
decide options parameteer is a list.
Not sure which one is better in C#, but isn't it better to make all consistent (all lists probably)?